backend/app/services/attachment/chunked_upload.py

```
@@ -0,0 +1,580 @@
+                )
+
+        # Store chunk
+        chunk_key = self._chunk_key(upload_id, chunk_index)
+        redis_client.setex(chunk_key, UPLOAD_EXPIRATION, chunk_data)
+
+        # Update session
+        if chunk_index not in session.received_chunks:
+            session.received_chunks.append(chunk_index)
+            session.received_chunks.sort()
+
+        session.chunk_checksums[chunk_index] = hashlib.md5(chunk_data).hexdigest()
```

Ruff (0.15.2)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)